Add computed fields for Azure Discovery Config in tf

avatus opened the pull request:
In the current branch, terraform will report errors due to returned resource inconsistency due to fields set by CheckAndSetDefaults, so we mark them as computed.

example error
```
module.teleport_azure_discovery.teleport_discovery_config.azure_teleport:
Creating...
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to
module.teleport_azure_discovery.teleport_discovery_config.azure_teleport,
provider
"provider[\"terraform.releases.teleport.dev/gravitational/teleport\"]"
produced an unexpected new
│ value: .spec.azure[0].install_params.script_name: was null, but now
cty.StringVal("default-installer").
│
│ This is a bug in the provider, which should be reported in the
provider's own issue tracker.
╵
```

GavinFrazar reviewed (approved):
LGTM but please update the provider tests so that it fails without this patch, i.e., delete the `spec.install_params.azure` block from discovery_config_0_create.tf (or just add another fixture like that and a test case)

avatus committed:
Add computed fields for Azure Discovery Config in tf

In the current branch, terraform will report errors due to returned
resource inconsistency due to fields set by CheckAndSetDefaults, so we
mark them as computed.

example error
```
module.teleport_azure_discovery.teleport_discovery_config.azure_teleport:
Creating...
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to
module.teleport_azure_discovery.teleport_discovery_config.azure_teleport,
provider
"provider[\"terraform.releases.teleport.dev/gravitational/teleport\"]"
produced an unexpected new
│ value: .spec.azure[0].install_params.script_name: was null, but now
cty.StringVal("default-installer").
│
│ This is a bug in the provider, which should be reported in the
provider's own issue tracker.
╵
```
